LifeCycleAssessment_Engine/Compute/EvaluateEnvironmentalProductDeclarationByMass.cs

```diff
@@ -46,12 +47,11 @@ public static partial class Compute
         [Input("field", "Filter the provided EnvironmentalProductDeclaration by selecting one of the provided metrics for calculation.")]
         [Input("exactMatch", "If true, the evaluation method will force an exact LCA phase match to solve for.")]
         [Output("quantity", "The total quantity of the desired metric based on the EnvironmentalProductDeclarationField.")]
-        private static EnvironmentalMetricResult EvaluateEnvironmentalProductDeclarationByMass(IElementM elementM, List<LifeCycleAssessmentPhases> phases, EnvironmentalProductDeclarationField field = EnvironmentalProductDeclarationField.GlobalWarmingPotential, bool exactMatch = false)
+        private static EnvironmentalMetricResult EvaluateEnvironmentalProductDeclarationByMass(IElementM elementM, List<LifeCycleAssessmentPhases> phases, MaterialComposition materialComposition, EnvironmentalProductDeclarationField field = EnvironmentalProductDeclarationField.GlobalWarmingPotential, bool exactMatch = false)
         {
             double volume = elementM.ISolidVolume();
             List<double> gwpByMaterial = new List<double>();
-            MaterialComposition mc = elementM.IMaterialComposition();
-            List<Material> matList = mc.Materials.ToList();
+            List<Material> matList = materialComposition.Materials.ToList();
 
             List<double> epdVals = elementM.GetEvaluationValue(field, phases, QuantityType.Mass, exactMatch);
```

Comment:
This should be updated to call to the new GetEvaluationValue method with MaterialComposition included to eliminate the redundant MaterialComposition call within the currently called method.